### PR TITLE
Newsim

### PR DESCRIPTION
**Pull request by josephineyates**
This PR creates the first attempt at simulation. 
Sorry it is fairly big but hopefully the code/structure is understandable enough.

### CODE REVIEW

**Comment by josephineyates** [comment]
Also a note, you can ignore the simulate_dataset.py file, it will be where I create the script to call from the CLI but it does not work yet

**Comment by pawel-czyz** [comment]
Regarding the (binary) `.DS_Store` files, I'd suggest adding them to `.gitignore`.

File: simulations/notebooks/Sampling-subclones.ipynb [hunk 1]
```diff
@@ -15,10 +15,38 @@
...
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Global seed set to 0\n"
+     ]
+    },
+    {
+     "ename": "AnnDataReadError",
```
Comment (pawel-czyz) [inline simulations/notebooks/Sampling-subclones.ipynb:30]: Would cleaning the outputs be feasible in this case? (If the notebook doesn't take enormous time to run, it's a good practice to remove the outputs, as the diffs will be minimized and one doesn't leak the derived data accidentally).

File: simulations/notebooks/Sampling-subclones.ipynb [hunk 2]
```diff
@@ -286,7 +314,12 @@
...
-   "version": "3.8.13"
+   "version": "3.9.10"
+  },
+  "vscode": {
+   "interpreter": {
+    "hash": "b0fa6594d8f4cbf19f97940f81e996739fb7646882a419484c72d19e05852a7e"
+   }
   }
  },
  "nbformat": 4,
```
Comment (pawel-czyz) [inline simulations/notebooks/Sampling-subclones.ipynb:325]: In the file below, cleaning outputs could also be beneficial. (Github doesn't want to show the content as the diff is too large).

File: simulations/simul/cna/clone.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+from lib2to3.pgen2.token import OP
```
Comment (pawel-czyz) [inline simulations/simul/cna/clone.py]: Why is this needed?

File: simulations/simul/cna/clone.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        # then the anchor profile will be [True, True, False]
+        self.anchor_profile = self._anchor_gain_profile()
+
+    def is_ancestral(self) -> bool:
+        """whether the subclone is an ancestral subclone"""
+        return self._ancestral
+
+    def _generate(self) -> np.ndarray:
+        """
+        returns the array of changes, either created from the ancestral clone or generated de novo
```
Comment (pawel-czyz) [inline simulations/simul/cna/clone.py:50]: It may be useful to add information about the returned shape of the array.

File: simulations/simul/cna/clone.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            if the clone is ancestral
+        """
+        if self._ancestral:
+            return self._CNVGenerator.generate_ancestral_subclone()
+        else:
+            return self._CNVGenerator.generate_child_subclone(self._ancestral_changes)
+
+    def _anchor_gain_profile(self) -> np.ndarray:
```
Comment (pawel-czyz) [inline simulations/simul/cna/clone.py]: The returned type is a `List[bool]`, rather than `np.ndarray`.

Also, I'm not sure what is the purpose of this, but I used previously boolean tuples for anchors, as they can be used as keys in dictionaries. (If you however plan to slice other arrays with this booleans, then array or list would be better).

**Review by pawel-czyz — APPROVED** [review]
Thanks!

File: simulations/notebooks/Sampling-subclones.ipynb (hunk 1 above)
Comment (josephineyates) [reply simulations/notebooks/Sampling-subclones.ipynb:30]: Yes sorry I just forgot but you are entirely right.

File: simulations/simul/cna/clone.py (hunk 3 above)
Comment (josephineyates) [reply simulations/simul/cna/clone.py]: it's not, thanks for the catch, VS tends to like to add imports when I miswrite stuff

File: simulations/simul/cna/clone.py (hunk 5 above)
Comment (josephineyates) [reply simulations/simul/cna/clone.py]: I cast to tuple this list when using it in the probability dist, but my idea was that it gave more leeway to how to use these in general. Thanks for the type catch!
